## 75% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 2
Expected to fail tests: 0
Passed tests: 6
```
```
IDs of failed tests: 740490,702843 
```
<h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix/close-confirmation-swap&PR_ID=21466&APK_NAME=21466.apk&TR_CASE_IDS=740490,702843">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3696657">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/')]` </div><div>Device 2: Wait for element `Button` for max 120s and click when it is available </div></blockquote></p>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:350: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/98824c2bc2b1424aa1cf1cf848ac94aa?auth=43c7d1220d65fe794870af58721be9de">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a6f579e4dfdc47ddbe1892bb7d2a5c70?auth=0855d8d3836febf7d3fc049271c92984">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3696662">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `network-dropdown` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/test_wallet.py:249: in test_wallet_balance_mainnet
    self.errors.verify_no_errors()
base_test_case.py:192: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 For the Ether the wrong value 0.0051 is shown, expected 0.0052 in total
E    For the Uniswap the wrong value 0.127 is shown, expected 0.627 in total
E    For the Ether the wrong value 0.0 is shown, expected 0.0001 on Arbitrum
E    For the Uniswap the wrong value 0.0 is shown, expected 0.5 on Arbitrum
```

<br/><br/></td></tr></tbody></table></details><h3>Passed tests (6)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3696656">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e1dbae0341304cc994d6c3a364cd4f75?auth=5f85a9dd68ce30850dc4c22cfc69b3c7#100">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a62d874f2a14409c97cd57a5243c66d4?auth=1582b1aa2b8db4ee226822be1e929728#91">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3696658">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8c432b6245de4eaea62b33db63490215?auth=31f9ca507bad21634e7dcbb3bb563091#89">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3696655">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8c432b6245de4eaea62b33db63490215?auth=31f9ca507bad21634e7dcbb3bb563091#57">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3696661">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3696660">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3696659">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table></details>

--- a/src/status_im/contexts/wallet/swap/events.cljs
+++ b/src/status_im/contexts/wallet/swap/events.cljs
@@ -17,6 +17,7 @@
  (fn [{:keys [db]} [{:keys [network asset-to-receive open-new-screen?] :as data}]]
    (let [{:keys [wallet]}       db
          test-networks-enabled? (get-in db [:profile/profile :test-networks-enabled?])
+         view-id                (:view-id db)
          account                (swap-utils/wallet-account wallet)
          asset-to-pay           (if (get-in data [:asset-to-pay :networks])
                                   (:asset-to-pay data)
@@ -30,12 +31,13 @@
      {:db (-> db
               (assoc-in [:wallet :ui :swap :asset-to-pay] asset-to-pay)
               (assoc-in [:wallet :ui :swap :asset-to-receive] asset-to-receive)
-              (assoc-in [:wallet :ui :swap :network] network'))
+              (assoc-in [:wallet :ui :swap :network] network')
+              (assoc-in [:wallet :ui :swap :launch-screen] view-id))
       :fx (if network'
             [[:dispatch [:wallet/switch-current-viewing-account (:address account)]]
              [:dispatch
               (if open-new-screen?
-                [:navigate-to :screen/wallet.setup-swap]
+                [:open-modal :screen/wallet.setup-swap]
                 [:navigate-to-within-stack
                  [:screen/wallet.setup-swap :screen/wallet.swap-select-asset-to-pay]])]
              [:dispatch [:wallet.swap/set-default-slippage]]]
@@ -293,7 +295,7 @@
                                                          :screen/wallet.swap-set-spending-cap
                                                          :screen/wallet.swap-confirmation)])
                                          (when-not approval-required?
-                                           (rf/dispatch [:wallet/select-account-tab :activity])
+                                           (rf/dispatch [:wallet/end-swap-flow])
                                            (debounce/debounce-and-dispatch
                                             [:toasts/upsert
                                              {:id   :swap-transaction-pending
@@ -419,3 +421,12 @@
                                   :last-request-uuid
                                   :approved-amount
                                   :approval-transaction-id)))})))
+
+(rf/reg-event-fx :wallet/end-swap-flow
+ (fn [{:keys [db]}]
+   (let [launch-screen (get-in db [:wallet :ui :swap :launch-screen])
+         address       (get-in db [:wallet :current-viewing-account-address])]
+     {:fx [(when (= launch-screen :wallet-stack)
+             [:dispatch [:wallet/navigate-to-account-within-stack address]])
+           [:dispatch [:wallet/fetch-activities-for-current-account]]
+           [:dispatch [:wallet/select-account-tab :activity]]]})))

--- a/src/status_im/navigation/screens.cljs
+++ b/src/status_im/navigation/screens.cljs
@@ -541,7 +541,8 @@
      :component wallet-swap-select-asset-to-pay/view}
 
     {:name      :screen/wallet.setup-swap
-     :options   {:insets {:bottom? true}}
+     :options   {:modalPresentationStyle :overCurrentContext
+                 :insets                 {:bottom? true}}
      :component wallet-swap-setup-swap/view}
 
     {:name      :screen/wallet.swap-propasal

